website/_dogfooding/clientModuleExample.ts

```diff
@@ -6,12 +6,33 @@
  */
 
 import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';
+import type {Location} from 'history';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-export function onRouteUpdate({location}: {location: Location}): void {
-  // console.log('onRouteUpdate', {location});
+export function onRouteUpdate({
+  location,
+  previousLocation,
+}: {
+  location: Location;
+  previousLocation: Location;
+}): void {
+  if (ExecutionEnvironment.canUseDOM) {
+    console.log(`onRouteUpdate (Fired before DOM repaints)
+Previous location: ${previousLocation.pathname}
+Current location: ${location.pathname}
+Current heading: ${document.getElementsByTagName('h1')[0]?.innerText}`);
+  }
 }
-
-if (ExecutionEnvironment.canUseDOM) {
-  // console.log('client module example log');
+export function onRouteDidUpdate({
```

Comment:
what about adding this but only for local/deploy previews? we could merge without having logs in prod

+ may be useful to also add onRouteUpdateDelayed too and factorize a bit?

Reply:
Yes, we would definitely revert this before merging.

> and may be useful to also add `onRouteUpdateDelayed` too and factorize a bit?

That lifecycle's a bit hard to emulate, but we can see what can be meaningfully done in that case (not retrieving DOM)